### DISCUSSION TITLE
Config Help

### DISCUSSION THREAD

**Discussion by Kwyntonix** [devcompl/TTController#152] [Q&A] [answered]
Hello! I couldn't find my particular issue in the discussions, but it has to do with what apps I need installed in order for TTController to connect properly. I'm new to pc so I hope I provide enough detail. I have the "Toughpower iRGB Plus 850w" thermaltake power supply. The apps I have are the "TT RGB Plus" and "Tt DPS G App", along with the TTController you made, and Razer Synapse. But the only thing showing up in Razer Synapse is the TT RGB Plus. I'm not sure if this is an issue, but I also don't have a physical in person TT Controller. I can control the rgb from the DPS G app, but nowhere else. I'm not sure where to go from here, but if you could help me out, that would be awesome!

**Comment by MoshiMoshi0** [comment]
You only need razer synapse and TTController. In TTController you would configure the detected DPSG controller to use the razer effect.
Since you only have the psu and want to use razer for rgb then you can just as well use thermaltake software.

Note that with TTController running you wont be able to use "Tt DPS G App" to check psu stats like voltage etc.

**Reply by Kwyntonix** [reply]
How do I configure the the DPSG controller to use the razer effect? The TTController isn't showing up in Synapse, and the only thermaltake apps I have installed is the TT DPS G app since TTController needs to control it.

**Reply by MoshiMoshi0** [reply]
> How do I configure the the DPSG controller to use the razer effect?

[Here](https://moshimoshi0.github.io/TTController/config/service) and [here](https://moshimoshi0.github.io/TTController/plugins/effects/razerconnecteffect) are the docs.

> The TTController isn't showing up in Synapse

It wont show on itself without configuring. Also iirc it will show as "TT RGB Plus" in synapse since it is pretending to be "TT RGB Plus".

> the only thermaltake apps I have installed is the TT DPS G app since TTController needs to control it.

I think you are confused about this project, TTController replaces official thermaltake apps and controls thermaltake devices directly without talking to other software, you cant have any other official thermaltake software running while TTController service is running.

**Reply by Kwyntonix** [reply]
I input the values into "config.json", and I can click "debug" and the rest of the buttons in the TTController, but it's still not showing up in Synapse for some reason, and I can't figure out why. Another "config" file popped up after I edited the "config.json" as well, but I don't think that's the issue.

**Reply by MoshiMoshi0** [reply]
What buttons? There is no GUI in TTController, its just a background service.

**Reply by Kwyntonix** [reply]
I mean the TTController works fine, with the different commands that you click "0,1,2,3,4" for. Nothing is showing up in Synapse.

**Reply by MoshiMoshi0** [reply]
Did you make a profile with `RazerConnectEffect` targeting your psu port identifier?

Read the [docs](https://moshimoshi0.github.io/TTController/) first on how to use this program, you cannot just throw random text into the config file and expect it to work.

Also post the whole output of the report from debug menu.

**Reply by Kwyntonix** [reply]
I'm trying to figure out how to post the report, but I don't know how to export it into a log or "copy paste" it into here. Sorry, I'm very inexperienced when it comes to this stuff.

**Reply by Kwyntonix** [reply]
![Capture](https://user-images.githubusercontent.com/80989570/112278958-5e768f80-8c51-11eb-8cf9-322a5f5a7c98.PNG)

**Reply by MoshiMoshi0** [reply]
what do you have in your `config.json` file?

**Reply by Kwyntonix** [reply]
I tried to go along the tutorial as best I could but I couldn't find what the port was. I hope this doesn't look too bad.



[config.json.zip](https://github.com/MoshiMoshi0/TTController/files/6210061/config.json.zip)

**Comment by MoshiMoshi0** [comment, marked as answer]
Your config looks ok, assuming it works then this will enable you to use razer synapse:

```json
{
  "Profiles": [
    {
      "Name": "PSU",
      "Ports": [
        [9802, 9001]
      ],
      "Effects": [
        {
          "Type": "RazerConnectEffect",
          "Config": {
            "Trigger": {
              "Type": "AlwaysTrigger"
            }
          }
        }
      ]
    }
  ],
  "PortConfigs": [
    {
      "Ports": [
        [9802, 9001],
      ],
      "Config": {
        "ColorModifiers": [
          {
            "Type": "LerpLedColorModifier"
          }
        ]
      }
    }
  ]
}
```

**Reply by Kwyntonix** [reply]
I tried that but it didn't seem to work. What do we do now? And I'll be sure to donate to your great cause <3

**Reply by MoshiMoshi0** [reply]
What didnt work.

**Reply by Kwyntonix** [reply]
It didn't show up in Synapse after using what you sent me. It didn't show up before though either.

**Reply by MoshiMoshi0** [reply]
What synapse version are you using?
Also, start the service in console mode using the above config and post the output.

**Reply by Kwyntonix** [reply]
[log.20210406.4.log](https://github.com/MoshiMoshi0/TTController/files/6262352/log.20210406.4.log)

**Reply by Kwyntonix** [reply]
I'm using ver 3.6.331.31613

**Reply by MoshiMoshi0** [reply]
The log says that the config is not getting loaded, you must save it in `config.json` file in the same directory as the exe file.
Make sure you enabled file extensions to be visible otherwise you might be saving as `config.json.json`

**Reply by Kwyntonix** [reply]
IT WORKS! I just had to copy and paste the config from `config.json` file into the regular `config` file. Thank you very much for baring with me, I know it must've been a headache. I can't wait to see what you do in the future with this app! In what currency should I send my donation to you?
